lib/r10k/git/rjgit.rb

```diff
@@ -60,6 +60,44 @@ def rev_parse(ver, opts = {})
         return commit.id
       end
 
+      def fetch(remote, opts={})
+
+        local = opts[:git_dir]
+        raise R10K::Git::GitError.new('fetch requires that opts[:git_dir] be a valid directory.') unless local && File.directory?(local)
+
+        repo = ::RJGit::Repo.new(local, is_bare: true)
+
+        rjgit_opts = opts.merge({refspecs: "+refs/*:refs/*"})
+        begin
+          repo.git.fetch(remote, rjgit_opts)
+        rescue Java::OrgEclipseJgitApiErrors::GitAPIException,
+               Java::OrgEclipseJgitApiErrors::TransportException,
+               Java::OrgEclipseJgitApiErrors::InvalidRemoteException => e
+          raise R10K::Git::GitError.new(e.message)
+        end
+
+        return true
+      end
+
+      def clone(remote, opts={})
+        local = opts[:git_dir]
+        raise R10K::Git::GitError.new('clone requires that opts[:git_dir] be a valid directory.') unless local && File.directory?(local)
+
+        rjgit_opts = opts.merge({branch: :all, is_bare: true})
+
+        repo = ::RJGit::Repo.new(local, is_bare: true)
```

Comment:
Seems kind of weird that you have to instantiate a repo object before cloning, but if that's the way of things then (shrug)....